operator/internal/controller/redpanda/redpanda_controller.go

```diff
@@ -621,27 +637,50 @@ func (r *RedpandaReconciler) configSyncMode(ctx context.Context, rp *redpandav1a
 	}
 }
 
-func (r *RedpandaReconciler) usersTXTFor(ctx context.Context, rp *redpandav1alpha2.Redpanda) (map[string][]byte, error) {
+func (r *RedpandaReconciler) superusersFor(ctx context.Context, rp *redpandav1alpha2.Redpanda) ([]string, error) {
 	values, err := rp.GetValues()
 	if err != nil {
 		return nil, errors.WithStack(err)
 	}
 
 	if !values.Auth.SASL.Enabled {
-		return map[string][]byte{}, nil
+		return nil, nil
 	}
 
 	key := client.ObjectKey{Namespace: rp.Namespace, Name: values.Auth.SASL.SecretRef}
 
 	var users corev1.Secret
 	if err := r.Client.Get(ctx, key, &users); err != nil {
 		if apierrors.IsNotFound(err) {
-			return map[string][]byte{}, nil
+			return nil, nil
 		}
 		return nil, errors.WithStack(err)
 	}
 
-	return users.Data, nil
+	superusers := []string{}
+	for filename, userTXT := range users.Data {
+		superusers = append(superusers, utils.LoadUsersFile(ctx, filename, userTXT)...)
+	}
+
+	// internal superuser should always be added
+	superusers = append(superusers, values.Auth.SASL.BootstrapUser.Username())
+
+	userProvidedSuperusers, ok := values.Config.Cluster["superusers"]
+	if ok {
+		switch userProvidedSuperusers := userProvidedSuperusers.(type) {
+		case []string:
+			superusers = append(superusers, userProvidedSuperusers...)
+		case []interface{}:
+			for _, su := range userProvidedSuperusers {
+				suStr, ok := su.(string)
+				if ok {
+					superusers = append(superusers, suStr)
+				}
+			}
+		}
+	}
+
+	return utils.NormalizeSuperusers(superusers), nil
```

Comment:
All of this logic should be handled in the cluster config syncer and I _think_ it is. Can we remove this and just return on line 666?